.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -12,4 +12,6 @@ Address any potentially confusing code. Is there code added that needs to be cle
 
  How is this change tested (unit tests, fuzz tests, etc.)? Are there any testing steps to be verified by the reviewer?
 
+ Is this a refactor change? If so, is the change tested using snapshot/before and after tests?
```

Comment:
Maybe add a more general "how have you proved that the behavior hasn't changed"?

Reply:
How about - `"Is this a refactor change? How have you proved that the intended behavior hasn't changed (Are you using snapshot/before and after tests)?`

Reply:
Sounds good to me, although the parentheses are probably unnecessary:

`Is this a refactor change? If so, how have you proved that the intended behavior hasn't changed? Are you using "snapshot" or "before and after" tests?`

Although, do "snapshot" or "before and after" tests have an accepted definition? Will people know what they are?

Reply:
Googling this, ended up at [pytest-snapshot](https://pypi.org/project/pytest-snapshot/); +1 on maybe a definition link or an example pattern we prefer.

Reply:
I don't think it makes sense to ask users to verify snapshots when we haven't provided any snapshots for them to verify results against. Today, we have to expect the user to answer "no" to the new questions. We should put the mechanism in place, and then remind developers to use the snapshot tests locally to avoid waiting for CI to execute the mechanism.

Reply:
@rday That is a good point, we could be more general now

```
Is this a refactor change? If so, how have you proved that the intended behavior hasn't changed? 
```

and once we have the snapshot tests (when [#2234](https://github.com/awslabs/s2n/issues/2234) is resolved) we can append to the template. 

```
Is this a refactor change? If so, how have you proved that the intended behavior hasn't changed? Are you using "snapshot" or "before and after" tests?
```